src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimePropertyInfo.cs

```diff
@@ -179,6 +179,13 @@ public override IList<CustomAttributeData> GetCustomAttributesData()
         public override Module Module => GetRuntimeModule();
         internal RuntimeModule GetRuntimeModule() { return m_declaringType.GetRuntimeModule(); }
         public override bool IsCollectible => m_declaringType.IsCollectible;
+
+        public override bool Equals(object? obj) =>
+            obj == (object)this ||
```

Comment:
Same as change to `ReferenceEquals` as https://github.com/dotnet/runtime/pull/73009/files#r935802561